kernel/kernel-api/src/test/scala/io/delta/kernel/internal/PaginationContextSuite.scala

```diff
@@ -182,11 +196,26 @@ class PaginationContextSuite extends AnyFunSuite {
       PaginationContext.forPageWithPageToken(
         TEST_TABLE_PATH,
         TEST_TABLE_VERSION,
+        TEST_LOG_SEGMENT_HASH,
         TEST_WRONG_PREDICATE_HASH,
         pageSize,
         validPageToken)
     }
     assert(e.getMessage.contains("Invalid page token: token predicate"))
   }
 
+  test("should throw exception for when the requested log segment doesn't " +
```

Comment:
Can we do an e2e test for this?

We could create a delta table with e.g. v2manfiest file A;
Then delete that file and add a new v2manifest file B

And our pagination logic should catch this